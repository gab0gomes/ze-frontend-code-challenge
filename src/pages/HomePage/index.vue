<template>
  <div>
    <h1>You are in HomePage</h1>
    <h2>This is the Foo component</h2>
    <address-input />
  </div>
</template>

<script>
import AddressInput from './components/AddressInputComponent'

export default {
  components: {
    AddressInput
  }
}
</script>
