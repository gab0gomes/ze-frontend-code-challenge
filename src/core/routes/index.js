import HomePage from '@/pages/HomePage'
import ProductsPage from '@/pages/ProductsPage'

const routes = [
  {
    path: '/',
    name: 'Home',
    component: HomePage
  },
  {
    path: '/products',
    name: 'Products',
    component: ProductsPage
  }
]

export default routes
