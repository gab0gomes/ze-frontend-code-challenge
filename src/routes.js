import HomePage from '@/pages/Home'

const routes = [
  {
    path: '/',
    name: 'Home',
    component: HomePage
  }
]

export default routes
