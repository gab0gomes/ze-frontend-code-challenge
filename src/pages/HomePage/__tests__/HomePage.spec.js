import { mount, createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'
import flushPromises from 'flush-promises'
import Multiselect from 'vue-multiselect'
import VueRouter from 'vue-router'

import HomePage from '../index'
import AddressSelect from '../components/AddressSelectComponent'

import addressStore from '@/core/store/address'
import geocodingStore from '@/core/store/requests/geocoding'

import routes from '@/core/routes'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)

const store = new Vuex.Store({
  modules: {
    address: {
      ...addressStore
    },
    requests: {
      namespaced: true,
      modules: {
        geocoding: {
          ...geocodingStore
        }
      }
    }
  }
})

const router = () =>
  new VueRouter({
    routes,
    linkActiveClass: 'active',
    mode: 'abstract'
  })

jest.mock('@/core/services/geocoding')

describe('HomePage', () => {
  it('render correctly', async () => {
    const wrapper = mount(HomePage, {
      localVue,
      store
    })

    expect(wrapper.element).toMatchSnapshot()

    wrapper.destroy()
  })

  it('should update address store after AddressSelect emit a select event', async () => {
    expect.assertions(1)
    jest.spyOn(store, 'commit')

    const wrapper = mount(HomePage, {
      localVue,
      store,
      router: router()
    })

    await flushPromises()

    const payload = {
      text: 'a',
      value: { lat: 1234, lng: 5432 }
    }

    const $addressSelect = wrapper.findComponent(AddressSelect)
    $addressSelect.vm.$emit('select', {
      ...payload
    })

    await flushPromises()

    expect(store.commit).toHaveBeenCalledWith(
      'address/setAddress',
      {
        fancyAddress: 'a',
        location: { lat: 1234, lng: 5432 }
      },
      undefined
    )

    wrapper.destroy()
  })

  it('should redirect to ProductsPage after AddressSelect emit a select event', async () => {
    expect.assertions(1)

    const routerInstance = router()
    jest.spyOn(routerInstance, 'push')

    const wrapper = mount(HomePage, {
      localVue,
      store,
      router: routerInstance
    })

    await flushPromises()

    const payload = {
      text: 'a',
      value: { lat: 1234, lng: 5432 }
    }

    const $addressSelect = wrapper.findComponent(AddressSelect)
    $addressSelect.vm.$emit('select', {
      ...payload
    })

    await flushPromises()

    expect(routerInstance.push).toHaveBeenCalledWith('/products')

    wrapper.destroy()
  })
})
