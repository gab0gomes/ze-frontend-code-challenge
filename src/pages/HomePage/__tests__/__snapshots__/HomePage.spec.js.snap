// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage render correctly 1`] = `
<main>
  <section
    class="container"
  >
    <div
      class="select-container"
    >
      <div
        class="multiselect"
        tabindex="-1"
      >
        <div
          class="multiselect__select"
        />
          
        <div
          class="multiselect__tags"
        >
          <div
            class="multiselect__tags-wrap"
            style="display: none;"
          />
           
          <!---->
           
          <transition-stub
            name="multiselect__loading"
          >
            <div
              class="multiselect__spinner"
              style="display: none;"
            />
          </transition-stub>
           
          <input
            autocomplete="nope"
            class="multiselect__input"
            name=""
            placeholder="Insira o endereço para ver produtos"
            style="width: 0px; position: absolute; padding: 0px;"
            tabindex="0"
            type="text"
          />
           
          <!---->
           
          <span
            class="multiselect__placeholder"
          >
            
          Insira o endereço para ver produtos
        
          </span>
        </div>
         
        <transition-stub
          name="multiselect"
        >
          <div
            class="multiselect__content-wrapper"
            style="max-height: 300px; display: none;"
            tabindex="-1"
          >
            <ul
              class="multiselect__content"
              style="display: block;"
            >
               
              <!---->
                
              <li
                style="display: none;"
              >
                <span
                  class="multiselect__option"
                >
                  Nenhum resultado encontrado. :(
                </span>
              </li>
               
              <li>
                <span
                  class="multiselect__option"
                >
                  Sem opções.
                </span>
              </li>
               
            </ul>
          </div>
        </transition-stub>
      </div>
       
      <!---->
    </div>
     
    <div
      class="typing-container"
    >
      <p
        class="line typing-animation"
      >
        
    Todes merecem uma bebida onde, como e quando quiserem!
  
      </p>
    </div>
  </section>
</main>
`;
